src/Application/Gallery/Gallery.test.tsx

```diff
@@ -1,21 +1,168 @@
+    const { asFragment } = render(<Component initialVisibility={true} />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should match snapshot when imagesDisplayed is set', () => {
+    const { asFragment } = render(<Component imagesDisplayed={3} />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('should match snapshot when item is less than defaultImagesDisplayed', () => {
+    const { asFragment } = render(
+      <GalleryItemLessThanDefaultImagesDisplayed />
+    );
+    expect(asFragment()).toMatchSnapshot();
+  });
+});
+
+describe('<Gallery /> open and close Slider modal', () => {
+  it('should show slider modal when a gallery item is clicked', () => {
+    render(<Component />);
+    const item = document.querySelectorAll('.gallery-item')[0];
+    expect(document.querySelector('.aries-slider')).not.toBeInTheDocument();
+    userEvent.click(item);
+    expect(document.querySelector('.aries-slider')).toBeVisible();
+  });
+
+  it('should close slider modal when close button is clicked', () => {
+    render(<Component initialVisibility={true} />);
+    expect(document.querySelector('.aries-slider')).toBeVisible();
+    const closeButton = screen.queryByTestId('close-button');
+    userEvent.click(closeButton);
+    expect(document.querySelector('.aries-slider')).not.toBeInTheDocument();
+  });
+});
+
+describe('<Gallery /> active item', () => {
+  test('click on gallery item', () => {
+    const { asFragment } = render(<Component />);
+    const itemIndex = 3;
+    const item = document.querySelectorAll('.gallery-item')[itemIndex];
+    userEvent.click(item);
+    const thumbnails = screen.queryAllByTestId('gallery_thumbnail');
+    expect(thumbnails[itemIndex]).toHaveClass('active');
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('click on gallery thumbnail', () => {
+    const { asFragment } = render(<Component initialVisibility={true} />);
+    const thumbnails = screen.queryAllByTestId('gallery_thumbnail');
+    expect(thumbnails[0]).toHaveClass('active');
+
+    const thumbnailIndex = 3;
+    const thumbnail = thumbnails[thumbnailIndex];
+    userEvent.click(thumbnail);
+    expect(thumbnails[thumbnailIndex]).toHaveClass('active');
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('click on slider next button', () => {
+    render(<Component />);
+    const itemIndex = 3;
+    const item = document.querySelectorAll('.gallery-item')[itemIndex];
+    userEvent.click(item);
+
+    const nextButton = screen.queryByTestId('slider_right-arrow');
+    userEvent.click(nextButton);
+
+    const thumbnails = screen.queryAllByTestId('gallery_thumbnail');
+    expect(thumbnails[itemIndex + 1]).toHaveClass('active');
+  });
+
+  test('click on slider prev button', () => {
+    render(<Component />);
+    const itemIndex = 3;
+    const item = document.querySelectorAll('.gallery-item')[itemIndex];
+    userEvent.click(item);
+
+    const nextButton = screen.queryByTestId('slider_left-arrow');
```

Comment:
Should be `prevButton` or `backButton`?

Reply:
haha right, should be `prevButton`, thank u!